### PR TITLE
Gh9015

### PR DESCRIPTION
**Pull request by IslandRhythms**

### CODE REVIEW

**Review by vkarpov15 — COMMENTED** [review]
Looking good so far. Next time, make sure to make a separate branch before creating a PR so the previous changes don't end up in the same PR :+1:

**Comment by IslandRhythms** [comment]
> Looking good so far. Next time, make sure to make a separate branch before creating a PR so the previous changes don't end up in the same PR 👍

Sorry about that, I had thought I made each one a separate branch.

File: lib/aggregate.js [hunk 1]
```diff
@@ -1105,14 +1105,9 @@ function prepareDiscriminatorPipeline(aggregate) {
     const discriminatorKey = discriminatorMapping.key;
     const discriminatorValue = discriminatorMapping.value;
 
-    // If the first pipeline stage is a match and it doesn't specify a `__t`
-    // key, add the discriminator key to it. This allows for potential
-    // aggregation query optimizations not to be disturbed by this feature.
-    if (originalPipeline[0] && originalPipeline[0].$match && !originalPipeline[0].$match[discriminatorKey]) {
```
Comment (vkarpov15) [inline lib/aggregate.js:1111]: Why did you remove this code? This change seems unintentional...

File: lib/schema.js [hunk 2]
```diff
@@ -400,21 +399,17 @@ Schema.prototype.defaultOptions = function(options) {
   const baseOptions = get(this, 'base.options', {});
   options = utils.options({
     strict: 'strict' in baseOptions ? baseOptions.strict : true,
-    strictQuery: 'strictQuery' in baseOptions ? baseOptions.strictQuery : false,
     bufferCommands: true,
     capped: false, // { size, max, autoIndexId }
     versionKey: '__v',
     optimisticConcurrency: false,
-    discriminatorKey: '__t',
```
Comment (vkarpov15) [inline lib/schema.js:408]: This change also seems unintentional. Deleting the `discriminatorKey` option is most definitely a non-goal

**Review by vkarpov15 — CHANGES_REQUESTED** [review]
